### DISCUSSION TITLE
Issues for new user with syncing

### DISCUSSION THREAD

**Discussion by Redgums** [abraunegg/onedrive#2546] [General]
I have a number of computers running Windows 10 or Debian 12 "Bookworm" on AMD64 and ARM64 systems. I am trying to use:

    abraunegg/onedrive: OneDrive Client for Linux
 
from github to keep the systems sync'ed. Obviously, this is seamless on Windows (as it should be), but the Linux systems are problematic for me.

All machines have the same basic directory structure:

Windows:
c:\Users\User1\
   Desktop
   Documents
   Download
   Music
   OneDrive
   Videos
   
Linux:
/home/user1/
   Desktop
   Documents
   Download
   Music
   OneDrive
   Videos

 I only want to sync the Documents directory so that it is the same on each system.
 
Problem is that I get unintended effects from my current configuration, and nothing I've tried fixes the problem. I've done various things using "skip_dir" and "skip_dotfiles"
 
 See below:
 ```
 sync_dir = "/home/user1/"
# skip_file = "~*|.~*|*.tmp"
monitor_interval = "300"
skip_dir = "OneDrive/"
skip_dir = ".cache/"
skip_dir = ".config/"
skip_dir = ".local/"
skip_dir = ".mozilla/"
skip_dir = ".pki/"
skip_dir = ".thunderbird/"
log_dir = "/home/user1/OneDrive/"  
skip_dotfiles = "true"
```
The issue is that when I use Firefox, Thunderbird, Transmission or download from the net, Onedrive gets conniptions and starts sending notifications about upload failure due to the file in ".local/...." not existing. 
```
Cannot move item: .cache/thunderbird/5qbk56ir.default-default/cache2/doomed/1012408364: No such file or directory
Received inotify delete event from operating system .. attempting item deletion as requested
Cannot upload file changes/creation: .thunderbird/5qbk56ir.default-default/storage.sqlite-journal: No such file or directory
Received inotify delete event from operating system .. attempting item deletion as requested
Cannot upload file changes/creation: .mozilla/firefox/zpkd3coe.default-release/prefs-1.js: No such file or directory
Cannot upload file changes/creation: .mozilla/firefox/zpkd3coe.default-release/prefs-1.js: No such file or directory
Cannot upload file changes/creation: .cache/thunderbird/5qbk56ir.default-default/cache2/entries/463A46855947C7642A8B047ADCA33B28D1F91589: No such file or directory
Cannot move item: .cache/thunderbird/5qbk56ir.default-default/cache2/doomed/888169460: No such file or directory
Received inotify delete event from operating system .. attempting item deletion as requested
Cannot upload file changes/creation: .thunderbird/5qbk56ir.default-default/storage/permanent/chrome/idb/3870112724rsegmnoittet-es.sqlite-wal: No such file or directory
Received inotify delete event from operating system .. attempting item deletion as requested
```
I believe that this is the various apps' cache usage triggering constant updates. As you can see, I've tried "skipping" the hidden files and folders created by these apps, but nothing seems to work.

Note: user is member of sudoers but is not running "sudo onedrive --monitor".

Can anyone point me in the right directions, please?

**Comment by abraunegg** [comment]
@Redgums 
>I only want to sync the Documents directory so that it is the same on each system.

The information point you are missing here is that Windows achieves this via using what is called 'Folder Redirection'. On Linux, this is called a symbolic link.

Your problem stems from this:
```
sync_dir = "/home/user1/"
log_dir = "/home/user1/OneDrive/"
```
*Do not do this sort of configuration ....*

On Linux, you have all sorts of other files and folder that should never be uploaded to OneDrive, including things like your SSH keys.

This is what you need to do:
* Revert every single configuration change you have made
* Disable any systemd service and/or other process that may be running 'onedrive'
* Configure the 'onedrive' application on Linux to have the *default* 'sync_dir' - `~/OneDrive/`
* Inside your `~/OneDrive` folder create a symbolic link from `~/OneDrive/Documents --> ~/Documents`
* Put your 'log_dir' somewhere else in your Linux Home Directors ... `~/logs/onedrive/` for example .....

Once re-configured .. 
* *check* your configuration with `--display-config`
* *test* manually using `--dry-run`

Once you are happy with what the application will do ... sync your data for real.

One last item ..... *do not run* `sudo onedrive ....` do not do this at any stage ..... it will run the client as 'root' ... this is *not* what you want to happen.
